### PR TITLE
Add placement Strategy

### PR DESCRIPTION
**Pull request by Hounddog**
Hi, 

when we started using this plugin we ended up having the slaves spread over all the EC2 Hosts.
However for proper utilization of our cluster we want to fill up the hosts before using a new one..

Our Setup:

- We have 3 Types of slaves running against our cluster
  - 30GB/20GB and 10GB Memory Reservation per slave
- For auto-scaling we have a Custom Lambda Script calculating if the Biggest Slave Container fits on any host

This change adds support for the ECS Placement Strategy https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-placement-strategies.html

### CODE REVIEW

**Comment by pbsladek** [comment]
@pgarbe Possible to get this in the 1.20 release?

**Comment by pgarbe** [comment]
Many thanks for the contribution @Hounddog. It's not part of v1.20 but we can schedule it for 1.21 @pbsladek 

I was wondering why it's only one task placement that can be set when ECS allows a [list of placement strategies](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-placement-strategies.html)?

**Comment by Hounddog** [comment]
@pgarbe Not sure what you mean by list of placement Strategies.

If you mean to have placement strategy also across availability zones?

Like here 
```
"placementStrategy": [
    {
        "field": "attribute:ecs.availability-zone",
        "type": "spread”
    },
    {
        "field": "memory",
        "type": "binpack"
    }
]
```

Maybee we can just have 2 fields to insert this kind of strategy?

**Comment by pgarbe** [comment]
What about adding a list of strategies (with field and type) like it is with env variables or logging configuration?

**Comment by Hounddog** [comment]
@pgarbe I guess that would be possible. I am not sure when i would be able to implement this change. As i am going to be out a few days

**Comment by webratz** [comment]
Hi,
just wanted to check if you already had time to work on the changes, so we maybe could get it released with the next version

**Comment by 22phuber** [comment]
Awesome work @Hounddog! 👍 
When will this be merged/released? I could use this feature for our jenkins ci builds 🙈

**Comment by webratz** [comment]
As soon someones can implement the change requested by @pgarbe and resolves the conflicts we can merge it :)

**Comment by Hounddog** [comment]
I finally got the time to clean this up and will update this today.

**Comment by Hounddog** [comment]
I have rebased this on the current master and added the Placement Strategy as requested.
However i have read in the RunTaskRequest there is a maximum of 5 Strategies to be added.

I am unsure how i can add this. I will test everything tomorrow if its working as  requested with our company test jenkins.

File: src/main/resources/com/cloudbees/jenkins/plugins/amazonecs/pipeline/ECSTaskTemplateStep/config.jelly [hunk 1]
```diff
@@ -51,6 +51,12 @@
   <f:entry name="networkMode" title="${%Network mode}" field="networkMode">
     <f:select />
   </f:entry>
+  <f:entry name="placementStrategyType" title="${%Placement Strategy Type}" field="placementStrategyType">
+    <f:select />
+  </f:entry>
+  <f:entry title="${%lacement Strategy Field}" field="placementStrategyField">
```
Comment (webratz) [inline src/main/resources/com/cloudbees/jenkins/plugins/amazonecs/pipeline/ECSTaskTemplateStep/config.jelly]: looks like a typo, missing `p`?

**Comment by rohitagarwal11** [comment]
When we are expecting to get this released?

**Comment by webratz** [comment]
Currently waiting on a response from @Hounddog if it works in their prod env, and a fix of the typo.
Also currently the PR build didn't work (see https://ci.jenkins.io/blue/organizations/jenkins/Plugins%2Famazon-ecs-plugin/detail/PR-81/12/pipeline )

**Comment by Hounddog** [comment]
Sorry for the late response... I was sick the last 2 weeks gonna continue today and fix the typo

**Comment by Hounddog** [comment]
I have it all working so far and tests are also running locally.
How would i get the test to pass in jenkins here?
It just seems that we might need to rerun the test

**Comment by Hounddog** [comment]
Please Rebuild

**Comment by webratz** [comment]
it seems that sth is not right with the PR builder currently :( i'll try to run it on my local machine and merge then

**Pull request merged**